code/obj/item/flamethrower.dm

```diff
@@ -79,6 +79,23 @@ A Flamethrower in various states of assembly
 		if(istype(src.gastank) && src.gastank.air_contents && istype(src.fueltank) && src.fueltank.reagents)
 			return TRUE
 
+
+	/// allow refilling the fuel tank by simply clicking the reagent dispensers
+	afterattack(atom/target, mob/user, flag)
+		if(istype(target, /obj/reagent_dispensers) && get_dist(src,target) <= 1)
+			if(istype(src.fueltank))
+				var/obj/o = target
+				o.reagents.trans_to(src.fueltank, (src.fueltank?.reagents?.maximum_volume - (src.fueltank?.reagents?.total_volume)))
+				inventory_counter?.update_percent(src.fueltank?.reagents?.total_volume, src.fueltank?.reagents?.maximum_volume)
```

Comment:
no need to nullcheck fueltank a million times, nor reagents - you already checked fueltank above and just wrap .reagents in an if/nif